ducktape/services/service.py

```diff
@@ -23,6 +23,38 @@
 import time
 
 
+class AbstractServiceIdFactory:
+    def generate_service_id(self):
+        raise NotImplementedError()
+
+
+class ServiceIdFactory(AbstractServiceIdFactory):
```

Comment:
You can make this object shared between instances (or even a singleton) if you make `generate_service_id()` method accept service as an argument.
Same for multirun factory, you can create it once per run in the runner_client rather than once per service per run.